src/nunavut/cli.py

```diff
@@ -15,10 +15,17 @@
 import sys
 import textwrap
 import typing
+import nunavut.lang
 
 
-def _should_generate_support(args: argparse.Namespace) -> bool:
-    if args.generate_support == 'as-needed':
+def _should_generate_support(
+    args: argparse.Namespace,
+    language_context: 'nunavut.lang.LanguageContext'
+) -> bool:
+    if language_context.get_target_language() is not None and \
```

Comment:
This has mypy errors but I don't think the logic is correct. commandline options should have a higher priority then internal language options